### PR TITLE
docs: contributing runbook

### PR DESCRIPTION
**Pull request by tisonkun**
Add a CONTRIBUTING file for inline guides. They're too details to be included in the Contribution Guide (or at least at the very first time). But I think when the content becomes stable and well-structured, we should publish them to the Contribution Guide and only leave the FAQs and links.

This closes https://github.com/apache/pulsar/issues/17900.

Signed-off-by: tison <wander4096@gmail.com>

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+### How to update reference pages?
+
+The source of reference pages consists of:
+
+* Generated from configuration classes
+* Generated from command-line interfaces
+* Manually held under `static/referces` folder
+
+Read [update reference docs](https://pulsar.apache.org/contribute/document-contribution/#update-reference-docs) guide for details.
```
Comment (Anonymitaet) [inline CONTRIBUTING.md:19]: Suggest consolidating this part into [update reference docs - guide](https://pulsar.apache.org/contribute/document-contribution/#update-reference-docs) to keep a single source of truth.

**Review by Anonymitaet — APPROVED** [review]
Thanks for your contribution! 

It's OK to add this info to https://pulsar.apache.org/contribute/ directly rather than here (put all relevant info in a centralized place). We can update this info later if needed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tisonkun) [reply CONTRIBUTING.md:19]: This section is covered by the "Update reference docs" guide. I add this CONTRIBUTING runbook for FAQs/How-tos and links to the related pages.

**Comment by tisonkun** [comment]
image

@Anonymitaet has moved to the contribution guide .

File: contribute/site-intro.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+## Pages
+
+Docusaurus provides three kinds of pages out-of-the-box: [docs](https://docusaurus.io/docs/docs-introduction), [blogs](https://docusaurus.io/docs/blog), and [JSX pages](https://docusaurus.io/docs/creating-pages).
+
+The Pulsar site pages are of:
+
+| Page                                       | Type      | Source                                                                                            |
+|--------------------------------------------|-----------|---------------------------------------------------------------------------------------------------|
+| [User docs](pathname:///docs)              | docs      | <ul><li>docs/</li><li>versioned_docs/</li><li>versioned_sidebars/</li><li>sidebars.json</li></ul> |
+| [Contrib guides](about.md)                 | docs      | <ul><li>contribute/</li><li>sidebarsDevelopment.js</li></ul>                                      |
```
Comment (Anonymitaet) [inline contribute/site-intro.md:20]:
```suggestion
| [Contribution guides](about.md)                 | docs      | <ul><li>contribute/</li><li>sidebarsDevelopment.js</li></ul>                                      |
```

File: contribute/site-intro.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+| [Blogs](pathname:///blog)                  | blog      | <ul><li>blog/</li></ul>                                                                           |
+| Other pages                                | JSX pages | <ul><li>src/pages/</li></ul>                                                                      |
+
+Besides, the site serves multiple static pages generated outside the framework, including API docs, reference docs, and swagger files. You can find them under the `static` folder.
+
+## Tools
+
+### preview.sh
+
+The most commonly used tools is `preview.sh`. You can preview your local changes by:
```
Comment (Anonymitaet) [inline contribute/site-intro.md:32]:
```suggestion
The most commonly used tool is `preview.sh`. You can preview your local changes by:
```

File: contribute/site-intro.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+To emulate the server-side logics, like `.htaccess` rewrite rules, you can run:
+
+```shell
+yarn build
+docker-compose up
+```
+
+### Pytools
+
+The site repo has a set of Python scripts for generating content and syncing/updating/publish the site.
```
Comment (Anonymitaet) [inline contribute/site-intro.md:53]:
```suggestion
The site repo has a set of Python scripts for generating content and syncing/updating/publishing the site.
```

File: contribute/site-intro.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+You can read the [README](https://github.com/apache/pulsar-site/tree/main/tools/pytools/README.md) file of pytools for details.
+
+## How-tos
+
+This section holds common how-tos about website maintenance and troubleshooting.
+
+### How to fix search index mismatches?
+
+First of all, you should get the permission to access `apache_pulsar` crawler on [Algolia Crawler console](https://crawler.algolia.com/). You can email dev@pulsar.apache.org to ask for permission.
```
Comment (Anonymitaet) [inline contribute/site-intro.md:63]:
```suggestion
First of all, you should get permission to access `apache_pulsar` crawler on [Algolia Crawler console](https://crawler.algolia.com/). You can email dev@pulsar.apache.org to ask for permission.
```

**Review by Anonymitaet — APPROVED** [review]

**Comment by tisonkun** [comment]
@Anonymitaet Thanks for your review! Integrated.

Pending to megre...

**Pull request merged**